### PR TITLE
[border-agent] update meshcop test cases

### PR DESCRIPTION
**Pull request by superwhd**
- Removed the verification on the meshcop service's name.
- Added a new test case to verify the meshcop service is published with proper fields. This test case will be extended when https://github.com/openthread/ot-br-posix/issues/736 is fixed. 

Related to: https://github.com/openthread/ot-br-posix/issues/735

### CODE REVIEW

File: tests/scripts/thread-cert/border_router/test_publish_meshcop_service.py [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+            'is_host': True
+        },
+    }
+
+    def test(self):
+        # TODO: verify the behavior when thread is disabled
+        host = self.nodes[HOST]
+        host.bash('service otbr-agent stop')
+        br = self.nodes[BR]
+        client = self.nodes[ROUTER]
```
Comment (wgtdkp) [inline tests/scripts/thread-cert/border_router/test_publish_meshcop_service.py:79]:
```suggestion
        host = self.nodes[HOST]
        br = self.nodes[BR]
        client = self.nodes[ROUTER]
        
        # TODO: verify the behavior when thread is disabled
        host.bash('service otbr-agent stop')
```

File: tests/scripts/thread-cert/border_router/test_publish_meshcop_service.py [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+        client.start()
+        self.simulator.go(5)
+        self.assertEqual('router', client.get_state())
+
+        self.check_meshcop_service()
+
+        br.disable_backbone_router()
+        self.check_meshcop_service()
+
+    def check_meshcop_service(self):
```
Comment (wgtdkp) [inline tests/scripts/thread-cert/border_router/test_publish_meshcop_service.py:97]:
```suggestion
    def check_meshcop_service(self, br, host):
```

**Review by wgtdkp — APPROVED** [review]
LGTM :+1:

**Comment by superwhd** [comment]
I've removed the dependency and resolved the conflict.

**Review by jwhui — APPROVED** [review]

**Pull request merged**